File: pinyougou-parent/pinyougou-manager-web/src/main/webapp/js/service/brandService.js
/*品牌管理服务层*/
app.service("brandService",function ($http) {
    /*查询全部数据*/
    this.findAll=function () {
        return $http.get('../brand/findAll.do');
    }
    /*分页查询数据*/
    this.findPage=function (page,size) {
        return $http.get('../brand/findPage.do?page='+page+'&size='+size);
    }
    /*新增数据*/
    this.add=function (entity) {
        return $http.post('../brand/add.do',entity);
    }
    /*更新数据*/
    this.update=function (entity) {
        return $http.post('../brand/update.do',entity);
    }
    /*查询单个数据*/
    this.findOne=function (id) {
        return $http.get('../brand/findOne.do?id='+id);
    }
    /*根据id数组删除数据*/
    this.delete=function (selectIds) {
        return $http.get('../brand/delete.do?ids='+selectIds);
    }
    /*条件查询数据*/
    this.search=function (page,size,searchEntity) {
        return $http.post('../brand/search.do?page='+page+'&size='+size,searchEntity);
    }
    
    /*根据下拉列表要求查询*/
    this.selectOptionList=function () {
        return $http.get('../brand/selectOptionList.do');
    }

});

File: pinyougou-parent/pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbBrand;
import entity.PageResult;

import java.util.List;
import java.util.Map;

/*
* 品牌接口
* */
public interface BrandService {
    /**
     * 查询所有记录
     * @return
     */
    public List<TbBrand> findAll();

    /**
     * 品牌记录分页
     * @param pageNum 当前页面
     * @param pageSize 每页记录数
     * @return
     */
    public PageResult findPage(int pageNum,int pageSize);

    /**
     *新增品牌
     * @param brand 品牌对象
     */
    public void add(TbBrand brand);

    /**
     *根据id查询品牌
     * @param id 品牌id
     * @return 品牌对象
     */
    public TbBrand findOne(long id);

    /**
     * 更新品牌
     * @param brand 品牌对象
     */
    public void update(TbBrand brand);

    /**
     *
     * @param ids id数组
     */
    public void delete(Long[] ids);

    /**
     * 根据要求查询品牌记录并分页
     * @param brand 将查询条件存入实体类
     * @param pageNum 当前页面
     * @param pageSize 每页记录数
     * @return 查询并分页后的数据
     */
    public PageResult search(TbBrand brand, int pageNum,int pageSize);

    /**
     *
     * @return 返回封装为map的品牌列表
     */
    public List<Map> selectOptionList();


}
